Adding the log to the required action to show the cause of syntax violation of the LDAP policy

### Description

It is requested that more traces appear at the "Caused by" level so that in the event of an incident, the problem description is clearer and the tool's administration team can act accordingly.

For example (the caused by message should be logged) :
``2025-10-09 09:25:14,567 WARN  [org.keycloak.services.resources.admin.UserResource] (executor-thread-18) Could not update user password.: org.keycloak.models.ModelException: Could not modify attribute for DN [cn=hsimpson@your.company.com,ou=Users,dc=your,dc=company,dc=com]
  at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.modifyAttributes(LDAPOperationManager.java:604)
  at org.keycloak.storage.ldap.idm.store.ldap.LDAPIdentityStore.updatePassword(LDAPIdentityStore.java:378)
  at org.keycloak.storage.ldap.LDAPStorageProvider.updateCredential(LDAPStorageProvider.java:870)
  at org.keycloak.credential.UserCredentialManager.updateCredential(UserCredentialManager.java:92)
  at org.keycloak.models.cache.infinispan.SubjectCredentialManagerCacheAdapter.updateCredential(SubjectCredentialManagerCacheAdapter.java:41)
  at org.keycloak.services.resources.admin.UserResource.resetPassword(UserResource.java:772)
  at org.keycloak.services.resources.admin.UserResource$quarkusrestinvoker$resetPassword_1adf3b6878cf142f29faf5dbd5c936893d03e559.invoke(Unknown Source)
  at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
  at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
  at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
  at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:638)
  at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
  at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
  at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
  at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
  at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
  at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
  at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
  at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: javax.naming.directory.InvalidAttributeValueException: [LDAP: error code 19 - Password fails quality checking policy]; remaining name 'cn=hsimpson@your.company.com,ou=Users,dc=your,dc=company,dc=com'
  at java.naming/com.sun.jndi.ldap.LdapCtx.mapErrorCode(LdapCtx.java:3273)
  at java.naming/com.sun.jndi.ldap.LdapCtx.processReturnCode(LdapCtx.java:3206)
  at java.naming/com.sun.jndi.ldap.LdapCtx.processReturnCode(LdapCtx.java:2997)
  at java.naming/com.sun.jndi.ldap.LdapCtx.c_modifyAttributes(LdapCtx.java:1505)
  at java.naming/com.sun.jndi.toolkit.ctx.ComponentDirContext.p_modifyAttributes(ComponentDirContext.java:277)
  at java.naming/com.sun.jndi.toolkit.ctx.PartialCompositeDirContext.modifyAttributes(PartialCompositeDirContext.java:192)
``


### Value Proposition

In the event of an incident, the Administration team has a clearer description of the problem and can act accordingly.

### Goals

The user should see the more detailed message while making a password change.

### Non-Goals

N/A

### Discussion

_No response_

### Notes

_No response_

--- a/services/src/main/java/org/keycloak/services/resources/admin/UserResource.java
+++ b/services/src/main/java/org/keycloak/services/resources/admin/UserResource.java
@@ -789,7 +789,11 @@ public class UserResource {
             logger.error(e.getMessage(), e);
             throw ErrorResponse.error(e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR);
         } catch (ModelException e) {
-            logger.warn("Could not update user password.", e);
+            String exceptionMessage = e.getCause() != null ? e.getCause().getMessage() : e.getMessage();
+            logger.warnf("Could not update password for user %s. Reason: %s", user.getUsername(), exceptionMessage);
+            if (logger.isTraceEnabled()) {
+                logger.trace("Could not update user password.", e);
+            }
             Properties messages = AdminRoot.getMessages(session, realm, auth.adminAuth().getToken().getLocale());
             throw new ErrorResponseException(e.getMessage(), MessageFormat.format(messages.getProperty(e.getMessage(), e.getMessage()), e.getParameters()),
                     Status.BAD_REQUEST);

